apollo-router/src/services/mod.rs

```diff
@@ -302,28 +302,45 @@ impl QueryPlannerRequest {
     pub fn new(
         query: String,
         operation_name: Option<String>,
-        query_plan_options: QueryPlanOptions,
         context: Context,
     ) -> QueryPlannerRequest {
         Self {
             query,
             operation_name,
-            query_plan_options,
+            query_plan_options: QueryPlanOptions::default(),
             context,
         }
     }
 }
 
 assert_impl_all!(QueryPlannerResponse: Send);
-/// [`Context`] and [`QueryPlan`] for the response..
+/// [`Context`] and [`QueryPlan`] for the response.
 pub struct QueryPlannerResponse {
     pub content: QueryPlannerContent,
     pub context: Context,
 }
 
+/// Query, QueryPlan and Introspection data in an opaque type.
+#[derive(Debug, Clone)]
+pub struct QueryPlannerContent(pub(crate) QueryPlannerContentInner);
+
+impl std::ops::Deref for QueryPlannerContent {
+    type Target = QueryPlannerContentInner;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
```

Comment:
`Deref` is a public API so this sort of defeats the point of this PR, even if what users can do with an enum value without being able to name the enum is very limited. Please remove this impl and add `.0` in internal uses as needed.